.semaphore/semaphore.yml

```diff
@@ -33,11 +34,15 @@ blocks:
       # don't run the tests on non-functional changes...
       when: "change_in('/', {exclude: ['/.deployed-versions/', '.github/']})"
     task:
+      env_vars:
+        - name: SEMAPHORE_AGENT_UPLOAD_JOB_LOGS
+          value: when-trimmed
       jobs:
         - name: Test
           commands:
             - . ci-tools ci-update-version
-            - mvn -Dmaven.gitcommitid.nativegit=true -U -Dmaven.wagon.http.retryHandler.count=10 --batch-mode --no-transfer-progress clean install
+            - mvn -Dsurefire.rerunFailingTestsCount=3 -Dmaven.gitcommitid.nativegit=true -U -Dmaven.wagon.http.retryHandler.count=10 --batch-mode --no-transfer-progress clean install
+            - cache store maven-$SEMAPHORE_GIT_BRANCH ~/.m2/repository
```

Comment:
If we are not using `maven-$SEMAPHORE_GIT_BRANCH ` key for restore, I think just `cache store` should be sufficient